storage.go

```diff
@@ -163,21 +161,14 @@ func (s *Storage) Boot(ctx context.Context) error {
 }
 
 func (s *Storage) Delete(ctx context.Context, k microstorage.K) error {
-	var err error
-
-	var body []byte
-	{
-		v := storageConfigJSONPatch{}
-		v.Spec.Storage.Data = map[string]*string{}
-		v.Spec.Storage.Data[k.Key()] = nil
-
-		body, err = json.Marshal(&v)
-		if err != nil {
-			return microerror.Mask(err)
-		}
+	storageConfig, err := s.g8sClient.CoreV1alpha1().StorageConfigs(s.namespace.Name).Get(s.name, apismetav1.GetOptions{})
+	if err != nil {
+		return microerror.Mask(err)
```

Comment:
We update the complete thing now. I did not come up with another solution since CRDs are still quite immature.